new-frontiers-14/frontier-station-14
# Round start spawn fixes, job display order and weights

## About the PR



Round start overflow jobs now check job requirements.  You can no longer spawn as a mercenary or pilot by joining early without meeting the required time.  Overflow jobs (contractor, pilot, merc) now spawn printing the correct station to the chat log.

Jobs have been given an overall spawn order.  Contractor, Pilot and Mercenary have been reduced to -1, and all other roles have been prioritized and given display weights as shown [in this spreadsheet (Google Docs)](https://docs.google.com/spreadsheets/d/1Gh6Qt7o9R_anN24F-eNCTrFUXBjSgUD9zs-7VsI9m5U/edit?usp=sharing)

Merc has been moved to the Civilian folder (it's in the Civilian department), and STC, SR, and Security Guard have been moved to a Frontier folder (again, same department).

**Known issues**: user priority/preference for contractor/pilot/merc spawning are not respected on round start if forced into an overflow job.

## Why / Balance


A new player could spawn as Pilot/Mercenary without meeting the time requirements if joining on round start.

Having a global order of weights helps round start jobs be filled in a sensible order if there are willing players, rather than everything being at zero.

## How to test


1. Connect as a new user to your local server (must have no time).
2. Press Ready, run `startround`.  You should spawn as a Contractor on Frontier, and your roundstart text should say so.
3. Run `golobby` to return to the lobby.
4. Repeat 2-3 a few times.
5. Add enough playtime to your character to play Pilot (`playtime_add_overall <account> <time>`), run `golobby`.
6. Press Ready, run `startround`.  You should spawn as Pilot on Exped Lodge 50% of the time, Pilot on Frontier 25% of the time, and Contractor on Frontier 25% of the time.
7. Run `golobby` to return to the lobby.
8. Repeat 6-7 a few times.
9. Add enough playtime to your character to play Mercenary (`playtime_add_overall <account> <time>`), run `golobby`.
10. Press Ready, run `startround`.  You should spawn as Pilot/Merc on Exped Lodge 25% of the time each, and Contractor, Pilot, and Merc on Frontier 16% of the time each.
11. Repeat 9-10 a few times.
12. Select any two station jobs at low/medium priority.  Hit Ready, run `startround`.  You should spawn as whichever job has the higher priority every time.
13. Put the lower priority job to High priority.  Hit ready, run `startround`.  You should spawn as the high priority job every time.

## Media


Job display order:
![image](https://github.com/user-attachments/assets/75b6ff29-7924-477f-a126-fb9d05e11b49)
![image](https://github.com/user-attachments/assets/aea8a563-8f68-4896-aa8b-6226d419c1f1)
![image](https://github.com/user-attachments/assets/e6e6feb6-59dc-4984-b16b-315bf2834107)



## Breaking changes


**Changelog**



Minor fix, no changelog.  Can add if needed.
